src/ed25519.cpp

```diff
@@ -86,6 +89,29 @@ bool verify(
             crypto_sign_ed25519_verify_detached(sig.data(), msg.data(), msg.size(), pubkey.data()));
 }
 
+std::pair<std::array<unsigned char, 32>, std::array<unsigned char, 64>>
+ed25519_pro_key_pair_for_ed25519_seed(std::span<const unsigned char> ed25519_seed) {
+    if (ed25519_seed.size() != 32)
+        throw std::invalid_argument{"Invalid ed25519_seed: expected 32 bytes"};
```

Comment:
This could also allow a 64-byte libsodium "secret key" as input (which is simply the seed+pubkey concatenated together, and so the first 32 bytes of that are still the seed).

Reply:
Done, added support for 32/64 byte keys